### ISSUE TITLE
Oracle Batch update

### ISSUE THREAD

**Issue by ralmsdeveloper** [aspnet/EntityFrameworkCore#9958]
Batch update, we have some limitations in Oracle, Example Return of the current transaction variable, in the insert using the cursor.

Are you thinking of implementing something like:
```sql
CREATE TYPE MICROSOFT IS TABLE of NUMBER;

DECLARE
OUTTABLE MICROSOFT; 
v_p0 NUMBER(10);
BEGIN
OUTTABLE := MICROSOFT();

INSERT INTO "Blog" ("Description") VALUES ('Test Sample 1') RETURNING "BlogId" INTO v_p0;
OUTTABLE.EXTEND(1);
OUTTABLE(OUTTABLE.COUNT) := v_p0;

INSERT INTO "Blog" ("Description") VALUES ('Test Sample 2') RETURNING "BlogId" INTO v_p0;
OUTTABLE.EXTEND(1);
OUTTABLE(OUTTABLE.COUNT) := v_p0; 

OPEN :cur for SELECT COLUMN_VALUE FROM TABLE(CAST(OUTTABLE AS MICROSOFT)); 
END;
```

I say this because in SQL Server, PostgreSQL and FirebirdSQL we do not have that inconvenience

I know you guys are doing their best.

**Comment by ajcvickers** [comment]
@ralmsdeveloper We're assuming that you are referencing the EF sample provider for Oracle here? If so, then these are our current general thoughts:
* We want to keep the sample working--we're setting up C.I. for this
* We will consider PRs from the community for the sample in the same way as for our other code
* We would like others to take this sample provider as a starting point and use it as the basis for a production-quality provider
  * If a production-quality open source provider becomes available, then we will likely reassess the need for the sample
* We're not intending to add any more functionality to the sample right now. Hopefully that will be done in the production-quality providers.

Coming back to your specific question, we're not intending to implement batching at the moment, but we would consider a PR that adds it. At the point there is a production-quality provider with basic support we will do what we can to help others with that, including guidance (if we have any) on supporting additional functionality.

**Comment by ralmsdeveloper** [comment]
@ajcvickers Thanks for the contact, I will study the case, I will try to contribute, I was seeing some situations.

**Comment by ajcvickers** [comment]
This is an issue identified on the Oracle sample provider. We don't plan to fix this in the sample, but we would be happy for a community contributor to fix it. Also, anyone taking the sample as a starting point for production code should consider fixing this issue.

**Comment by ralmsdeveloper** [comment]
Hello @ajcvickers,

I believe that issue can be closed, this was resolved with PR #10078
